components/scream/scripts/gen-boiler

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+"""
+Generate boiler-plate for converting a fortran subroutine to cxx.
+
+This script is designed to facilitate f90->cxx conversion of p3 and shoc
+subroutines, with the potential to be expanded to additional packages later.
+This script should also help maintain the latest approved coding patterns.
+
+To test this tool, run:
+% python3 -m doctest gen_boiler.py
+"""
+
+from utils import expect, check_minimum_python_version, get_git_toplevel_dir
+check_minimum_python_version(3, 5)
+
+import argparse, sys, pathlib
+
+from gen_boiler import GenBoiler, get_supported_pieces, get_supported_physics, get_piece_description
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} <subroutine-name> [<subroutine-name> ...] [--piece] [--kernel] [--verbose]
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Generate all boilerplate p3 subroutine foo_bar \033[0m
+    > {0} foo_bar
+
+    \033[1;32m# Generate all boilerplate for shoc subroutine bar_baz and func_name \033[0m
```

Comment:
I think these helps are a bit confusing: this one says it creates stuff for shoc, the next for p3, but there's no mention of pr or shoc in the help. One (e.g., me) may wonder how does gen-boiler figure out where to put bar_baz...

Reply:
Edit: I tried adding `--physics shoc`, and it works. Perhaps we should add that in the help.

Reply:
Done.

Reply:
Yeah, that solves a big confusion of mine. Thanks.